Update Readme and gitignore (#16)
Satograms/Satogram: README.md

## Custom Macaroon

You may want to use a macroon with more limited permissions.  You can prepare one with the following command

```shell
lncli bakemacaroon --save_to ./satogram.macaroon \
  uri:/lnrpc.Lightning/AddInvoice \
  uri:/invoicesrpc.Invoices/SubscribeSingleInvoice \
  uri:/lnrpc.Lightning/ListInvoices \
  uri:/lnrpc.Lightning/LookupInvoice \
  uri:/lnrpc.Lightning/SubscribeInvoices \
  uri:/lnrpc.Lightning/DescribeGraph \
  uri:/lnrpc.Lightning/GetInfo \
  uri:/lnrpc.Lightning/GetNetworkInfo \
  uri:/lnrpc.Lightning/GetNodeInfo \
  uri:/lnrpc.Lightning/QueryRoutes \
  uri:/routerrpc.Router/SendPaymentV2
```

And then use the satogram.macaroon in place of admin.macaroon

## Using the REST Server

Run with `go run cmd/rest_http_server/main.go --config path/to/config.yaml`

## LND Lightning Payments and Invoices Structure

In an LND node, the lightning invoices received from a keysend will include fields in the custom_records field of the following TLV types according to the [TLV registry](https://github.com/satoshisstream/satoshis.stream/blob/main/TLV_registry.md)

- 34349334 : The common keysend Chat message of variable length
- 5482373484 : The preimage for the invoice

> - 6789998212 : A new TLV type that will include a lightning address when the recipient pubkey is recognized as a custodial provider

Reviewer: ```suggestion
- 6789998212 : A new TLV type that will include a unique customer identifier (lightning address, cashtag, account username etc.), to be used by the custodial provider to identify who the payment is intended for
```

## For Custodial Providers

Multiple keysends may be sent over the lightning network to your Lightning Node Pubkey(s).  You can identify the account to credit it to based on the lightning address specified in the field "6789998212" if present within the custom records.

> For example, a value of "7765617279646F6F7235384077616C6C65746F667361746F7368692E636F6D0A" for the field "6789998212" maps to "wearydoor58@walletofsatoshi.com"

Reviewer: ```suggestion
For example, a value of  `7765617279646F6F7235384077616C6C65746F667361746F7368692E636F6D0A` for the field "6789998212" maps to "wearydoor58@walletofsatoshi.com"
```

## Viewing Payments

> The custom_records structure is included in teh last hop of the route. You can review it with the following command

Reviewer: ```suggestion
The custom_records structure is included in the last hop of the route. You can review it with the following command:
```

```sh
lncli listpayments | jq '.payments[].htlcs[].route.hops[-1].custom_records'
```

Sample Result

```json
{
  "34349334": "676d2066726f6d20746162636f6e66212040626974636f696e636f646572626f62",
  "5482373484": "20b1c195fc250c2e9aa4a1f6ee1b14fcf2b12285ed86d6cb360088af7e861e44",
  "6789998212": "656e6f7567687769726539384077616c6c65746f667361746f7368692e636f6d"
}
```

> You can get the raw string back by piping the hexadecimal representation through xxd

Reviewer: ```suggestion
You can get the raw string back by piping the hexadecimal representation through xxd:
```

```sh
echo "676d2066726f6d20746162636f6e66212040626974636f696e636f646572626f62" | xxd -ps -r | xargs echo
```

Result:
```sh
gm from tabconf! @bitcoincoderbob
```

A sample of using golang to decode hexadecimal strings is available in the following [Go Playground](https://go.dev/play/p/IbX2QolRkbp)

> ## Viewing Messages Received on LND

Reviewer: ```suggestion
## Viewing Messages Received in LND
```
